Add option to disable power-assert - fixes #1017

jfmengels opened the pull request:
Add option to disable power-assert - fixes #1017

Let me know if I've missed anything or if there are better ways to handle this.

cc @kentcdodds

<!--

Read the [contributing guidelines](https://github.com/avajs/ava/blob/master/contributing.md). We are excited about pull requests, but please try to limit the scope, provide a general description of the changes, and remember, it's up to you to convince us to land it. If this fixes an open issue, link to it in the following way: `Fixes #321`. New features and bug fixes should come with tests.

-->

sindresorhus commented on cli.js:
```diff
@@ -64,6 +64,7 @@ var cli = meow([
 	'  --tap, -t          Generate TAP output',
 	'  --verbose, -v      Enable verbose output',
 	'  --no-cache         Disable the transpiler cache',
+	'  --no-power-assert  Disable the power-assert feature',
```
> Disable Power Assert

kentcdodds commented on test/babel-config.js:
```diff
@@ -50,7 +50,7 @@ test('uses babelConfig for babel options when babelConfig is an object', functio
 	var fixtureSource = fs.readFileSync(fixturePath, 'utf8');
 
-	var options = babelConfigHelper.build(babelConfig, fixturePath, fixtureSource);
+	var options = babelConfigHelper.build(babelConfig, undefined, fixturePath, fixtureSource);
```
I'd rather give this `undefined` a variable name (like `var disablePowerAssert = undefined`) so people looking at this later know what it's for.

sindresorhus replied:
Might be worth making it an options object instead of arguments.

kentcdodds commented on test/babel-config.js:
```diff
@@ -89,3 +89,26 @@ test('should reuse existing source maps', function (t) {
+	var fixturePath = fixture('es2015.js');
+	var fixtureSource = fs.readFileSync(fixturePath, 'utf8');
+
+	var options = babelConfigHelper.build(babelConfig, true, fixturePath, fixtureSource);
```
Similarly here: `var disablePowerAssert = true`

kentcdodds commented:
Looks awesome!

sindresorhus commented on cli.js:
```diff
@@ -130,6 +131,7 @@ var api = new Api({
 	serial: cli.flags.serial,
 	require: arrify(cli.flags.require),
 	cacheEnabled: cli.flags.cache !== false,
+	disablePowerAssert: cli.flags.powerAssert === false,
```
Double-negatives are usually bad, better to make it `powerAssert: cli.flags.powerAssert !== false,`

kentcdodds replied:
agreed. So on the CLI it'd be:

> ava --power-assert=false

Or

> ava --power-assert false

?

jfmengels replied:
The CLI handler seems to handle `no-` well, to it sets `powerAssert` to false when usng `no-power-assert`. I think we should keep this.

sindresorhus replied:
Yeah, the CLI part should be kept as-is. `--no` is the best way to handle negative flags.

jfmengels commented:
I addressed all comments, except https://github.com/avajs/ava/pull/1024#discussion_r76693217 (let me know if you want to me to do it), and renamed disablePowerAsssert to powerAssert throughout my changes in the codebase.

kentcdodds commented:
LGTM. Thanks for working on this @jfmengels!

jfmengels commented:
The tests work on my end on Node v6.4.0. Any hints to make those failing tests pass? :confused:

sindresorhus commented:
@jfmengels Tests pass on 6.4.0, but not on the just released 6.5.0. So it seems something changed there that broke some assumptions in the tests. Unrelated to your changes though, as it breaks in master too.

sindresorhus commented:
Looks great! Congrats on your first pull request as a core team member :D

![leo-celebrate](https://cloud.githubusercontent.com/assets/170270/18110250/228f8032-6f17-11e6-8da5-3e872b3105fe.gif)

kentcdodds commented:
Great work @jfmengels! Thanks for working on that!
